app/models/concerns/test_track/identity.rb

```diff
@@ -46,6 +46,17 @@ def test_track_identifier(identifier_type, identifier_value_method)
             end
           end
         end
+
+        define_method :test_track_sign_up! do
+          discriminator = TestTrack::IdentitySessionDiscriminator.new(self)
+          identifier_value = send(identifier_value_method)
+
+          if discriminator.participate_in_unauthenticated_session?
+            discriminator.controller.send(:test_track_session).sign_up! identifier_type, identifier_value
+          else
+            TestTrack::OfflineSession.create_visitor_for(identifier_type, identifier_value)
```

Comment:
I guess it doesn't do anything terrible, but it doesn't help much, because any later assignment done from the model layer will lazily create a visitor for this identifier and sign it up too. Maybe it should noop.